### PR TITLE
Fix display of keyer messages containing embedded NL/CR characters

### PR DESCRIPTION
**Pull request by dl1jbe**
Furthermore clean up 'sendbuf()' and keyer terminal display

### CODE REVIEW

File: src/clear_display.c [hunk 1]
```diff
@@ -48,20 +48,68 @@
...
+    mvaddstr(4, 0, terminal4);
+}
+
+/*
+ * add 'buffer' to the keyer terminal while scrolling one line up
+ * and refreshing the display
+ */
+static void add_line_to_keyer_terminal(char *buffer) {
+
+    if (terminal1)
```
Comment (zcsahok) [inline src/clear_display.c:87]: just a minor note: g_free does the NULL check internally.

Comment (dl1jbe) [reply src/clear_display.c]: Thanks for the reminder.

File: src/clear_display.c [hunk 2]
```diff
@@ -48,20 +48,67 @@
...
+    clear_display();
+}
+
+/*
+ * add 'buffer' to keyer terminal, splitting it into separate lines if needed
+ */
+void add_to_keyer_terminal(char *buffer) {
+    gchar **lines = g_strsplit_set(buffer, "\n\r", 0);
+    for (int i = 0; lines[i] != NULL; i++)
+        add_line_to_keyer_terminal(lines[i]);
```
Comment (zcsahok) [inline src/clear_display.c:106]: Using this setting
```
F1= CQ CQ DE % % %|CQ CQ DX % % % K
```
and calling CQ twice Tlf shows this
image

whereas fldigi shows
image

There seem to be a leading CR in the digi keyer strings which leads to an empty line on Tlf keyer terminal.
How about moving the chomp'ing into the for loop before calling `add_line_to_keyer_terminal` and skipping if line is empty? This of course means not displaying embedded CRs ("text1|||text2"), but we have just 4 lines anyway...

Comment (dl1jbe) [reply src/clear_display.c:106]: Good point. I agree - the extra newline has to be dropped. Question is if we drop embedded newlines / carriage returns completely or cut them down to at most one newline.

Will look into it tomorrow. Today are other things to do :-) here.

Comment (dl1jbe) [reply src/clear_display.c]: Actual solution is as recommended - just drop empty lines from display. They will still be sent via the digit keyer.

As far as I remember the discussion some years ago digi mode users are very seldom use more newlines than really needed. They try to use vertical space very sparingly - maybe leftover from paper printers :-).

Comment (zcsahok) [reply src/clear_display.c]: Thanks, works OK.
HNY!

**Pull request merged**